### PR TITLE
Update authoring samples to .NET 6

### PR DESCRIPTION
**Pull request by angelazhangmsft**
Updating authoring samples to .NET6/VS2022, and modifying the docs/readmes. 

Also adding C++ WinAppSDK app to the AuthoringDemo sample.

### CODE REVIEW

File: src/Samples/AuthoringDemo/AuthoringDemo.sln [hunk 1]
```diff
@@ -1,44 +1,74 @@
...
+		{96752AD7-C344-4D4D-9E0B-39BED88B191A}.Release|ARM64.Build.0 = Release|ARM64
 		{96752AD7-C344-4D4D-9E0B-39BED88B191A}.Release|x64.ActiveCfg = Release|x64
 		{96752AD7-C344-4D4D-9E0B-39BED88B191A}.Release|x64.Build.0 = Release|x64
 		{96752AD7-C344-4D4D-9E0B-39BED88B191A}.Release|x86.ActiveCfg = Release|Win32
 		{96752AD7-C344-4D4D-9E0B-39BED88B191A}.Release|x86.Build.0 = Release|Win32
+		{7E449DC2-5334-464D-B478-1715B96C8050}.Debug|ARM64.ActiveCfg = Debug|arm64
+		{7E449DC2-5334-464D-B478-1715B96C8050}.Debug|ARM64.Build.0 = Debug|arm64
+		{7E449DC2-5334-464D-B478-1715B96C8050}.Debug|x64.ActiveCfg = Debug|x64
+		{7E449DC2-5334-464D-B478-1715B96C8050}.Debug|x64.Build.0 = Debug|x64
+		{7E449DC2-5334-464D-B478-1715B96C8050}.Debug|x86.ActiveCfg = Debug|x64
```
Comment (j0shuams) [inline src/Samples/AuthoringDemo/AuthoringDemo.sln:48]: Mismatch in architecture. Make sure BuildConfiguration is set up right for the solution

File: src/Samples/AuthoringDemo/WinUI3CppApp/MainWindow.xaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿<Window
+    x:Class="WinUI3CppApp.MainWindow"
+    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    xmlns:local="using:WinUI3CppApp"
+    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+    mc:Ignorable="d">
+
+    <StackPanel Orientation="Horizontal" HorizontalAlignment="Center" VerticalAlignment="Center">
```
Comment (j0shuams) [inline src/Samples/AuthoringDemo/WinUI3CppApp/MainWindow.xaml:10]: Are we intentionally not using the custom types here until the WinAppSDK scenario is fixed?

**Review by j0shuams — APPROVED** [review]
Looks good, thanks Angela! Just one thing about the build configurations for AuthoringDemo.sln

File: src/Samples/AuthoringDemo/AuthoringDemo.sln (hunk 1 above)
Comment (angelazhangmsft) [reply src/Samples/AuthoringDemo/AuthoringDemo.sln]: fixed

**Pull request merged**